### ISSUE TITLE
[Feature Suggestion] Upsolving Tracker

### ISSUE THREAD

**Issue by shreyg-upta** [C0ldSmi1e/Training-Tracker#1]
**Problem Statement:**  
Users often forget to revisit and solve problems they couldn't complete during the training session. This leads to missed learning opportunities and reduced effectiveness of our training tracker.  

**Proposed Solution:**  
1. **Upsolving Reminder:**  
   - After completing a training session, whenever a user visits the "Training" or "Statistics" tab, automatically check if there are unsolved problems from the previous session, in the background.  
   - If unsolved problems are detected, display a message in the "Training" tab:  
     _"We recommend upsolving previous problems before starting the next session."_  
   - Provide the following options for the user:  
     - **Upsolve Now**: Redirect the user to the problem.  
     - **Later**: Postpone the reminder until the next visit.  
     - **Give Up**: Mark the problem as skipped and remove it from reminders.  

2. **Statistics Enhancement:**  
   - In the "Statistics" tab, add a visual indicator (e.g., a yellow `-` symbol) next to problems that were upsolved after the session.  

**Benefits:**  
- Encourages users to complete unsolved problems, enhancing learning outcomes.  
- Improves user experience by making it easier to track progress and pending tasks.  
- Adds a motivational element by visually highlighting completed tasks in the statistics tab.

**Comment by C0ldSmi1e** [comment]
Thanks, will try doing it.

**Comment by shreyg-upta** [comment]
Do you want me to work on this?

**Comment by C0ldSmi1e** [comment]
I implemented the upsolve part, please feel free to test it.

But I have a new sub-page for upsolving, and the code is not the best practice, I am sure (the state management became a bit complicated at this point).

You can also improve the code quality or implement other parts of your requests.

Thanks a lot! 🍻
